chart/templates/_helpers.tpl

```diff
@@ -0,0 +1,787 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/*
+Get the user defined LoadBalancerIP for this release.
+Note, returns 127.0.0.1 if using ClusterIP.
+*/}}
+{{- define "carto.serviceIP" -}}
+{{- if eq .Values.router.service.type "ClusterIP" -}}
+127.0.0.1
+{{- else -}}
+{{- .Values.router.service.loadBalancerIP | default "" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Gets the host to be used for this application.
+If not using ClusterIP, or if a host or LoadBalancerIP is not defined, the value will be empty.
+*/}}
+{{- define "carto.baseUrl" -}}
+{{- $host := include "carto.serviceIP" . -}}
+
+{{- $port := "" -}}
+{{- $servicePortString := printf "%v" .Values.router.service.ports.http -}}
+{{- if and (not (eq $servicePortString "80")) (not (eq $servicePortString "443")) -}}
+  {{- $port = printf ":%s" $servicePortString -}}
+{{- end -}}
+
+{{- $defaultUrl := "" -}}
+{{- if $host -}}
+  {{- $defaultUrl = printf "%s%s" $host $port -}}
+{{- end -}}
+
+{{- default $defaultUrl (printf "%s" .Values.onPremDomain) -}}
+{{- end -}}
+
+{{/*
+Return the proper Carto ldsApi image name
+*/}}
+{{- define "carto.ldsApi.image" -}}
+{{- include "common.images.image" (dict "imageRoot" .Values.ldsApi.image "global" .Values.global) -}}
+{{- end -}}
+
+{{/*
+Return the proper Carto lds-api full name
+*/}}
+{{- define "carto.ldsApi.fullname" -}}
+{{- printf "%s-lds-api" (include "common.names.fullname" .) -}}
```

Comment:
This is potentially problematic.. You should truncate to 63 characters (+trimming "-" suffixes) to avoid DNS issues when the release name

Reply:
This concern applies to every "carto.XXX.fullname" template defined here.

Reply:
Done